core/jobs/batch_jobs/voiceover_migration_job.py

```diff
@@ -0,0 +1,314 @@
+
+
+class PopulateManualVoiceoversToEntityVoiceoversModelJob(base_jobs.JobBase):
+    """Migrates voiceovers stored in the exploration model to the new
+    EntityVoiceovers storage models.
+    """
+
+    DATASTORE_UPDATES_ALLOWED = True
+
+    @classmethod
+    def create_entity_voiceovers_model_instance(
+        cls,
+        entity_voiceovers_instance: voiceover_domain.EntityVoiceovers
+    ) -> voiceover_models.EntityVoiceoversModel:
+        """Creates an entity voiceovers model instance.
+
+        Args:
+            entity_voiceovers_instance: EntityVoiceovers. An instance of the
+                entity voiceovers domain class.
+
+        Returns:
+            EntityVoiceoversModel. An instance of the entity voiceover model.
+        """
+        entity_voiceovers_dict = entity_voiceovers_instance.to_dict()
+
+        entity_id = entity_voiceovers_dict['entity_id']
+        entity_type = entity_voiceovers_dict['entity_type']
+        entity_version = entity_voiceovers_dict['entity_version']
+        language_accent_code = entity_voiceovers_dict['language_accent_code']
+        voiceovers = entity_voiceovers_dict['voiceovers']
+
+        with datastore_services.get_ndb_context():
+            entity_voiceovers_model = (
+                voiceover_models.EntityVoiceoversModel.create_new(
+                    entity_type, entity_id, entity_version,
+                    language_accent_code, voiceovers
+                )
+            )
+        entity_voiceovers_model.update_timestamps()
+        return entity_voiceovers_model
+
+    @classmethod
+    def create_entity_voiceovers_instances(
+        cls,
+        exploration_model: exp_models.ExplorationModel,
+        exploration_voice_artists_link_model: (
+            voiceover_models.ExplorationVoiceArtistsLinkModel),
+        voice_artist_metadata_models_list: List[
+            voiceover_models.VoiceArtistMetadataModel]
+    ) -> List[voiceover_domain.EntityVoiceovers]:
+        """Creates a list of EntityVoiceovers instances containing voiceovers
+        from the given exploration data.
+
+        Args:
+            exploration_model: ExplorationModel. An instance of an
+                exploration model.
+            exploration_voice_artists_link_model:
+                ExplorationVoiceArtistsLinkModel. An instance of an
+                ExplorationVoiceArtistsLinkModel model.
+            voice_artist_metadata_models_list: list(VoiceArtistMetadataModel). A
+                list of VoiceArtistMetadataModel instances. Language accent code
+                for a given voice artist should be fetched from this model.
+
+        Returns:
+            list(EntityVoiceovers). A list of EntityVoiceovers instances
+            containing voiceovers from the given exploration data.
+
+        Raises:
+            Exception. All voice artists are not assigned accents in the
+                VoiceArtistMetadataModel.
+        """
+
+        voice_artist_id_to_language_code_mapping = {}
```

Comment:
Why don't we generate this just once at the outset of the job, and pass it ready-made into this function, rather than compute it again and again for each exploration?